mlir/lib/Driver/CompilerDriver.cpp

```diff
@@ -47,11 +51,121 @@
 #include "Quantum/Transforms/Passes.h"
 
 #include "Enzyme.h"
+#include "Timer.hpp"
 
 using namespace mlir;
 using namespace catalyst;
 using namespace catalyst::driver;
 
+namespace catalyst::utils {
+
+/**
+ * LinesCount : A utility class to count the number of lines of embedded programs
+ * in different compilation stages.
+ *
+ * You can dump the program-size embedded in an `Operation`, `ModuleOp`, or
+ * `llvm::Module` using the static methods in this class.
+ *
+ * To display results, run the driver with the `ENABLE_DEBUG_INFO=ON` variable.
+ * To store results in YAML format, use `DEBUG_RESULTS_FILE=/path/to/file.yml`
+ * along with `ENABLE_DEBUG_INFO=ON`.
```

Comment:
I'll have a look at the full PR soon (looks really good!), but one thing that doesn't quite sound right is the "debug" terminology. I don't think the information we produce is for debugging purposes, and we should keep those concepts distinct.

Although to be honest I don't have a good idea for the variables name right now 🤔 Maybe "diagnostics" related? Or "instrumentation" related, that's also the term used by MLIR for the pass instrumentation we are using.

Reply:
Absolutely! What about these env variables?
- `ENABLE_TIME_PROFILING` 
- `ENABLE_DIAGNOSTICS`
- `PROFILING_RESULTS_PATH`